packages/libs/hostapi/crypto-api.ts

```diff
@@ -87,7 +88,30 @@ export class CryptoApi {
     return resRaw.body;
   }
 
-  public ecRecover(data: Uint8Array): Uint8Array {
+  /**
+   * recover the address associated with the public key from elliptic curve signature or return zero on error.
+   * The function parameters correspond to ECDSA values of the signature:
+   * @param hash
+   * @param v final 32 bytes of signature
+   * @param r  first 32 bytes of signature
+   * @param s  second 32 bytes of signature
+   *
+   * @returns string returns an address, and not an address payable
+   */
+  public ecRecover(hash: string, v: BigInt, r: BigInt, s: BigInt): string {
+   if (v.countBits()!=256 || r.countBits()!=256 && s.countBits()!=256){
+     return ""
+   }
+    //[msgHash 32B][v 32B][r 32B][s 32B]
+    const syscallInput = hash
+        + v.toString(16)
```

Comment:
BigInt.toString(16) does not guarantee the string to be property padded to 32 bytes, extra checks might be needed for this.